biothings/hub/dataload/dumper.py

```diff
@@ -296,18 +296,28 @@ def register_status(self, status, transient=False, **extra):
                 )
                 self.src_doc.pop(field)
 
-        self.src_doc.update(
-            {
-                '_id': self.src_name,
-                'download': {
-                    'release': release,
-                    'data_folder': data_folder,
-                    'logfile': self.logfile,
-                    'started_at': datetime.now().astimezone(),
-                    'status': status,
-                },
+        current_download_info = {
+            '_id': self.src_name,
+            'download': {
+                'release': release,
+                'data_folder': data_folder,
+                'logfile': self.logfile,
+                'started_at': datetime.now().astimezone(),
+                'status': status
             }
-        )
+        }
+        # Update last success download time.
+        # If current status is success, we will get the current's started_at
+        # If failed, we will get the last_success from the last download instead.
+        last_download_info = self.src_doc.setdefault("download", {})
+        current_download_info["download"]["last_success"] = last_download_info.get("last_success")
+        if status == "success":
+            current_download_info["download"]["last_success"] = current_download_info["download"][
+                "started_at"
+            ]
+
```

Comment:
optionally, you could add a block of `else` here: if status is not success and `last_success` does not exist yet, you can then check `last_download_info` and set `last_success` if last one is a "success".

Also we should avoid set `last_success` to `None`, it should not be set in this case.

Reply:
got it, let's me update the code